frontendUiTableImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/GitMacheteGraphTableManager.java

```diff
@@ -92,55 +94,59 @@ private void subscribeToGitRepositoryChanges() {
   }
 
   @Override
-  @UIEffect
   public void refreshGraphTable() {
-    Option<GitRepository> gitRepository = gitRepositorySelectionProvider.getSelectedRepository();
-    if (gitRepository.isDefined()) {
-      Path macheteFilePath = getMacheteFilePath(gitRepository.get());
-      boolean isMacheteFilePresent = Files.isRegularFile(macheteFilePath);
+    synchronized (gitRepositorySelectionProvider) {
+      Option<GitRepository> gitRepository = gitRepositorySelectionProvider.getSelectedRepository();
+      if (gitRepository.isDefined()) {
+        Path macheteFilePath = getMacheteFilePath(gitRepository.get());
+        boolean isMacheteFilePresent = Files.isRegularFile(macheteFilePath);
 
-      refreshGraphTable(macheteFilePath, isMacheteFilePresent);
+        refreshGraphTable(macheteFilePath, isMacheteFilePresent);
+      }
     }
   }
 
   /** Creates a new repository graph and sets it to the graph table model. */
-  @UIEffect
   private void refreshGraphTable(Path macheteFilePath, boolean isMacheteFilePresent) {
-    LOG.debug(() -> "Entering: macheteFilePath = ${macheteFilePath}, isMacheteFilePresent = ${isMacheteFilePresent}");
-    // isUnitTestMode() checks if IDEA is running as a command line applet or in unit test mode.
-    // No UI should be shown when IDEA is running in this mode.
-    if (!project.isInitialized() || ApplicationManager.getApplication().isUnitTestMode()) {
-      LOG.debug("Project is not initialized or application is in unit test mode. Returning.");
-      return;
-    }
+    SwingUtilities.invokeLater(() -> {
```

Comment:
Isn't it enough to invoke on UI thread only lines 137-138 + 147-148 ?

Reply:
A similar approach has been taken on #257